### ISSUE TITLE
Android Build Config failing Unused Imports rule

### ISSUE THREAD

**Issue by matthewbahr-clear** [pinterest/ktlint#3022]
Note: I recognize this is a _very_ specific case. Not bothered if it's closed as wont do. 

## Expected Behavior

Create a new ktlint suppression or rule (however you want to frame it) allowing for a BuildConfig import in Android to be ignored when it breaks the import rules. 

## Observed Behavior

With this code snippet in our MainActivity:

```kt
package com.clearme.clearapp.home

// Android Studio fails to find this Config file usually, but it compiles fine at build time.
import com.clearme.clearapp.home.BuildConfig 

//Later in the Activity
`specificFeatureInitializer.initWithActivity(this@MainActivity, BuildConfig.DEBUG)`
```

The import line fails ktlint check, and running ktlint format removes it, which is incorrect behavior. 

Unfortunately the only possible fix is to suppress `no-unused-imports` for the entire file. 

`@file:Suppress("ktlint:standard:no-unused-imports")`

This might be a particular/special case for Android. The BuildConfig files are often missing until a full build runs, and even then they're often linking wrong (Even when marking generated sources root correctly AS fails to register the BuildConfig file - there are endless Stack Overflow complaints about this and any solution seems to be temporary).

Obviously this is problematic when it comes to enforcing ktlint style on the rest of the imports for a file. It can be partially ameliorated by isolating build config dependencies and injecting them (something we're migrating to but it's a slow process), but that just scopes the failing import down to the provider module that is creating the config. 

## Your Environment

* Version of ktlint used: 1.3.0, ktlint plugin 12.1.1
* Relevant parts of the `.editorconfig` settings
Nothing particularly relevant but here's what we're running:

```
[*.{kt,kts}]
end_of_line = lf
ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
ij_kotlin_imports_layout = *, java.**, javax.**, kotlin.**, ^
indent_size = 2
indent_style = space
ktlint_code_style = ktlint_official
ktlint_function_signature_body_expression_wrapping = multiline
ktlint_function_signature_rule_force_multiline_when_parameter_count_greater_or_equal_than = 2
ktlint_ignore_back_ticked_identifier = true
ktlint_function_naming_ignore_when_annotated_with=Composable
max_line_length = 140
```

* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `./gradlew ktlintCheck`
* Version of Gradle used (if applicable): 8.5.2
* Operating System and version: N/A

**Comment by paul-dingemans** [comment]
Tnx for reporting. You code code sample is too much simplified. I am not able to reproduce your problem. Also you're not running the latest version of ktlint. Please try that to check whether it resolves the problem.

FWIW, I am inclined to remove this rule from ktlint. It gives too many problems which can not be resolved by ktlint, as all lint decisions have to be made based on information in the file which contains the import. Ktlint is not able to do crosschecks with other files like the compiler does.

**Comment by matthewbahr-clear** [comment]
Sorry the code sample is too simplified - can't really give more. I think this is a compound issue of Gradle's generated code linking issues (which is part of why they're killing off the BuildConfig in Gradle 9) getting caught by ktlint/ 

Makes sense about removing it from ktlint - that definitely makes me more comfortable leaving it disabled for the activity.

**Comment by paul-dingemans** [comment]
Starting from Ktlint `1.7.0` the `no-unused-imports` rule will not be run by default anymore. See #3038
